provider/interface.go

```diff
@@ -6,12 +6,38 @@ import (
 	"github.com/pivotal-cf/brokerapi/domain"
 )
 
-//go:generate counterfeiter -o fakes/fake_service_provider.go . ServiceProvider
-type ServiceProvider interface {
-	Provision(context.Context, ProvisionData) (dashboardURL, operationData string, isAsync bool, err error)
-	Deprovision(context.Context, DeprovisionData) (operationData string, isAsync bool, err error)
-	Bind(context.Context, BindData) (binding domain.Binding, err error)
-	Unbind(context.Context, UnbindData) (unbinding domain.UnbindSpec, err error)
-	Update(context.Context, UpdateData) (operationData string, isAsync bool, err error)
-	LastOperation(context.Context, LastOperationData) (state domain.LastOperationState, description string, err error)
+type Provider interface{}
+
+type Updater interface {
+	Update(context.Context, UpdateData) (res *domain.UpdateServiceSpec, err error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o fakes/fake_provisioner.go . Provisioner
+type Provisioner interface {
+	Provision(context.Context, ProvisionData) (res *domain.ProvisionedServiceSpec, err error)
+	Deprovision(context.Context, DeprovisionData) (res *domain.DeprovisionServiceSpec, err error)
+}
+
+type AsyncProvisioner interface {
+	Provisioner
+	LastOperation(context.Context, LastOperationData) (state *domain.LastOperation, err error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o fakes/fake_binder.go . Binder
+type Binder interface {
+	Bind(context.Context, BindData) (binding *domain.Binding, err error)
```

Comment:
Does it make sense to have a Binder which is not also a Provisioner?
```suggestion
	Provisioner
	Bind(context.Context, BindData) (binding *domain.Binding, err error)
```

Reply:
Agreed ... I was also considering these too (some kind of internal struggle between "interfaces just do one thing" and "this is the real world, get over it" :smile: )

Reply:
ok, this gets a bit complicated and ends up a bit rabit holey ... I attempted to make Provider==Provisioner and embed the relevent types together, however it is quite a big change because:

* Avoiding overlapping interfaces becomes a problem (ie you want to implement Provider and AsyncBinder but NOT AsyncProvider)
* Some of the tests become significant;y harder to write (requiring major refactoring to accommodate - especially the api_tests which as mainly un-changed by this PR)

It's possible, but after a good 40mins I am backing out as the changes just keep piling up and I don't think the reward is worth the effort.

If you feel very strongly about these interface changes, then I can plough on, but it was alot more effort than expected.